pkg/server/admin.go

```diff
@@ -3543,6 +3546,50 @@ func (c *adminPrivilegeChecker) requireViewActivityAndNoViewActivityRedactedPerm
 	return nil
 }
 
+// requireViewClusterMetadataPermission requires the user have the VIEWCLUSTERMETADATA
+// system privilege and returns an error if the user does not have it.
+func (c *adminPrivilegeChecker) requireViewClusterMetadataPermission(
+	ctx context.Context,
+) (err error) {
+	userName, isAdmin, err := c.getUserAndRole(ctx)
+	if err != nil {
+		return serverError(ctx, err)
+	}
+	if !isAdmin {
+		hasViewClusterMetadata := false
+		if c.st.Version.IsActive(ctx, clusterversion.SystemPrivilegesTable) {
+			hasViewClusterMetadata = c.checkHasSystemPrivilege(ctx, userName, privilege.VIEWCLUSTERMETADATA)
+		}
+		if !hasViewClusterMetadata {
```

Comment:
I think this should go into the `IsActive` since if it's not active, we can't actually grant `VIEWCLUSTERMETADATA`